recipe/thirdparty/providers/googleWorkspaces.go

```diff
@@ -7,9 +7,12 @@ import (
 	"github.com/supertokens/supertokens-golang/supertokens"
 )
 
-func GoogleWorkspaces(input tpmodels.ProviderInput) tpmodels.TypeProvider {
-	if input.ThirdPartyID == "" {
-		input.ThirdPartyID = "google-workspaces"
+func GoogleWorkspaces(input tpmodels.ProviderInput) *tpmodels.TypeProvider {
+	if input.Config.ThirdPartyId == "" {
+		input.Config.ThirdPartyId = "google-workspaces"
+	}
+	if input.Config.Name == "" {
+		input.Config.Name = "Google Workspaces"
 	}
 
 	if input.Config.ValidateIdTokenPayload == nil {
```

Comment:
The error message below is "return errors.New("invalid id token payload")" in case hd doesn't match. Thats a bad error message